web-gateway/app/controllers/ItemController.scala

```diff
@@ -129,6 +129,39 @@ class ItemController(messagesApi: MessagesApi, userService: UserService, itemSer
     }
   }
 
+  def updateItemForm(itemId: UUID) = Action.async{ implicit rh =>
+    requireUser(user => loadNav(user).flatMap { implicit nav =>
+      val itemFuture = itemService.getItem(itemId)
+        .handleRequestHeader(authenticate(user)).invoke()
+      for( item <- itemFuture ) yield{
+        val (duration, durationUnits) = (item.auctionDuration.getSeconds match{
```

Comment:
You don't need a for comprehension here. You can directly map over the future returned by `invoke()`  (which is what the for comprehension is doing under the covers anyway).